lib/core/src/Cardano/Wallet.hs

```diff
@@ -1406,6 +1390,9 @@ joinStakePool ctx wid (pid, pools) argGenChange pwd = db & \DBLayer{..} -> do
     pure (tx, txMeta, txTime)
   where
     db = ctx ^. dbLayer @s @k
+    checkAlreadyJoined = \case
+        (WalletDelegation (Delegating pid') _) -> pid' == pid
+        _ -> False
```

Comment:
I am thinking if this is ok. Probably ok as first approximation. I just did not know how we want to treat situations like : we delegated to pId1, then can we quit as it is in the next? not active? Let's assume if yes, then what about situation joined to pId1, not active, then quit, then rejoined pId1. Etc. As I do not have this situations clear I opted for the simplest option here